ToDoList/ToDoList/Controls/ToDoList.ascx.cs

```diff
@@ -0,0 +1,163 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Web.UI;
+using System.Web.UI.WebControls;
+using ToDoList.Data;
+
+namespace ToDoList.Controls
+{
+    public partial class ToDoList : System.Web.UI.UserControl
+    {
+        public int NumberOfRecordsToDisplay { get; set; } = 5;
+        public string CategoryFilter
+        {
+            get
+            {
+                object o = ViewState["Category"];
+                if (o != null)
+                {
+                    return o.ToString();
+                }
+                else
+                {
+                    return "ALL";
+                }
+            }
+            set
+            {
+                ViewState["Category"] = value;
+            }
+        }
+
+        private List<ToDoItem> ListToDisplay;
+        private bool EnableNext = false;
+        private int CurrentIndex
+        {
+            get
+            {
+                object o = ViewState["CurrentIndex"];
+                if (o != null)
+                {
+                    return (int)o;
+                }
+                else
+                {
+                    return 0;
+                }
+            }
+            set
+            {
+                ViewState["CurrentIndex"] = value;
+            }
+        }
+
+        protected void Page_Load(object sender, EventArgs e)
+        {
+            if (!IsPostBack)
+            {
+                // Bind Categories list to CategoriesDropDown and set the initial selected value
+                List<string> categoriesList = ToDoItemData.GetCategories();
+                categoriesList.Add("ALL");
+                CategoryDropDown.DataSource = categoriesList;
+                CategoryDropDown.DataBind();
+                int currentCategoryIndex = ToDoItemData.GetCategories().IndexOf(CategoryFilter);
+                CategoryDropDown.SelectedIndex = currentCategoryIndex;
+            }
+        }
+
+        protected void Page_PreRender(object sender, EventArgs e)
+        {
+            BindListToDisplay();
+            NextButton.Enabled = EnableNext;
+            PreviousButton.Enabled = CurrentIndex == 0 ? false : true;
+        }
+
+        /// <summary>
+        /// Sets the Done property of a ToDoItem to true.
+        /// </summary>
+        /// <param name="source"></param>
+        /// <param name="e"></param>
+        protected void TODOs_ItemCommand(object source, RepeaterCommandEventArgs e)
+        {
+            int index = int.Parse(e.CommandArgument.ToString());
+            //List<ToDoItem> todos = ToDoItemData.GetToDoItems();
+            PruneList();
+            ListToDisplay[index].Done = true;
+            BindListToDisplay();
```

Comment:
Since you're calling the `BindListToDisplay` method in the `Page_PreRender` method, you can remove this call.